analyzers/src/SonarAnalyzer.CSharp/Rules/ShiftDynamicNotInteger.cs

```diff
@@ -21,30 +21,19 @@
 namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public sealed class ShiftDynamicNotInteger : ShiftDynamicNotIntegerBase<ExpressionSyntax>
+    public sealed class ShiftDynamicNotInteger : ShiftDynamicNotIntegerBase<SyntaxKind, ExpressionSyntax>
     {
-        private static readonly DiagnosticDescriptor rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
-
-        protected override DiagnosticDescriptor Rule { get; } = rule;
+        protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
 
         protected override bool ShouldRaise(SemanticModel semanticModel, ExpressionSyntax left, ExpressionSyntax right) =>
-            IsDynamic(left, semanticModel) &&
-            !IsConvertibleToInt(right, semanticModel);
+            IsDynamic(left, semanticModel) && !IsConvertibleToInt(right, semanticModel);
 
-        protected override bool CanBeConvertedTo(ExpressionSyntax expression, ITypeSymbol type, SemanticModel semanticModel)
-        {
-            var conversion = semanticModel.ClassifyConversion(expression, type);
-            return conversion.Exists && (conversion.IsIdentity || conversion.IsImplicit);
-        }
+        protected override bool CanBeConvertedTo(ExpressionSyntax expression, ITypeSymbol type, SemanticModel semanticModel) =>
+            semanticModel.ClassifyConversion(expression, type) is { Exists: true } conversion
+            && (conversion.IsIdentity || conversion.IsImplicit);
 
-        private static bool IsDynamic(ExpressionSyntax expression, SemanticModel semanticModel)
-        {
-            var type = semanticModel.GetTypeInfo(expression).Type;
-            return type != null && type.TypeKind == TypeKind.Dynamic;
-        }
+        private static bool IsDynamic(ExpressionSyntax expression, SemanticModel semanticModel) =>
+            semanticModel.GetTypeInfo(expression).Type is { TypeKind: TypeKind.Dynamic };
 
         protected override void Initialize(SonarAnalysisContext context)
```

Comment:
Sorry, I might have not been precise about what I meant to do. By rule registration, I meant the registration of the actions that are executed. Now that the rule uses the LanguageFacade it should be possible to move the Initialize method to the base class from both the C# and VB.NET files.